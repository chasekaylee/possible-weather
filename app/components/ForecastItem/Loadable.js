/**
 *
 * Asynchronously loads the component for ForecastItem
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
