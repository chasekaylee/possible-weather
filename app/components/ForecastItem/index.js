/**
 *
 * ForecastItem
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import CardContent from '@material-ui/core/CardContent';
import Card from '@material-ui/core/Card';

import WeatherIcon from 'components/WeatherIcon/Loadable';

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  paper: {
    height: 200,
    width: 200,
  },
  controlBar: {
    display: 'flex',
    flexDirection: 'row',
    margin: '2em',
    justifyContent: 'center',
  },
  control: {
    padding: theme.spacing(2),
  },
  cardContent: {
    flexGrow: 1,
  },
  weatherContent: {
    textAlign: 'center',
  },
  weatherContainer: {
    display: 'flex',
    justifyContent: 'center',
  },
}));

const formatTime = (time, format = 'dddd, MMMM Do') =>
  moment(time).format(format);

function ForecastItem(props) {
  const classes = useStyles();

  return (
    <Grid item key={`${props.day.time}`}>
      <Card key={`${props.day.time}`} className={classes.card}>
        <CardContent className={classes.cardContent}>
          <Grid key={`${props.day.time}`} item xs>
            <div className={classes.weatherContent}>
              <Typography component="h2" variant="h5">
                {formatTime(props.day.time * 1000, 'ddd')}
              </Typography>
              <div>
                <WeatherIcon name={props.day.icon} width="50" height="50" />

                <Typography gutterBottom variant="h5" component="h2">
                  <strong>{Math.round(props.day.temperatureHigh)}°</strong>
                  <div>{Math.round(props.day.temperatureLow)}°</div>
                </Typography>
              </div>
            </div>
          </Grid>
        </CardContent>
      </Card>
    </Grid>
  );
}

ForecastItem.propTypes = {
  // TODO: define shape
  day: PropTypes.object.isRequired,
};

export default memo(ForecastItem);
