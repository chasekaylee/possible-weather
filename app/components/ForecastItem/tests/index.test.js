/**
 *
 * Tests for ForecastItem
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ForecastItem from '../index';

const day = {
  time: 1583737200,
  summary: 'Clear throughout the day.',
  icon: 'clear-day',
  sunriseTime: 1583764440,
  sunsetTime: 1583806080,
  moonPhase: 0.52,
  precipIntensity: 0.0005,
  precipIntensityMax: 0.0021,
  precipIntensityMaxTime: 1583737560,
  precipProbability: 0.1,
  precipType: 'rain',
  temperatureHigh: 47.68,
  temperatureHighTime: 1583795640,
  temperatureLow: 30.58,
  temperatureLowTime: 1583846400,
  apparentTemperatureHigh: 45,
  apparentTemperatureHighTime: 1583795700,
  apparentTemperatureLow: 26.45,
  apparentTemperatureLowTime: 1583847420,
  dewPoint: 28.17,
  humidity: 0.74,
  pressure: 1024.1,
  windSpeed: 3.61,
  windGust: 6.8,
  windGustTime: 1583755320,
  windBearing: 9,
  cloudCover: 0.07,
  uvIndex: 3,
  uvIndexTime: 1583785260,
  visibility: 10,
  ozone: 387.9,
  temperatureMin: 27.68,
  temperatureMinTime: 1583763720,
  temperatureMax: 47.68,
  temperatureMaxTime: 1583795640,
  apparentTemperatureMin: 26.69,
  apparentTemperatureMinTime: 1583755140,
  apparentTemperatureMax: 45,
  apparentTemperatureMaxTime: 1583795700,
};

describe('<ForecastItem />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ForecastItem day={day} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<ForecastItem day={day} />);
    expect(firstChild).toMatchSnapshot();
  });
});
