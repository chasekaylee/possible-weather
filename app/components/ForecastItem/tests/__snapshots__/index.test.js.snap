// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastItem /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-item"
>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContent-175"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <div
          class="makeStyles-weatherContent-176"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5"
          >
            Mon
          </h2>
          <div>
            <img
              alt="clear-day"
              height="50"
              src="IMAGE_MOCK"
              width="50"
            />
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              <strong>
                48
                °
              </strong>
              <div>
                31
                °
              </div>
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
